fendermint/vm/topdown/src/sync.rs

```diff
@@ -0,0 +1,211 @@
+use ipc_agent_sdk::message::ipc::ValidatorSet;
+use ipc_sdk::cross::CrossMsg;
+use ipc_sdk::subnet_id::SubnetID;
+use std::sync::Arc;
+use std::time::Duration;
+
+/// Constantly syncing with parent through polling
+pub struct PollingParentSyncer {
+    config: Config,
+    parent_view_provider: Arc<InMemoryFinalityProvider>,
+    agent: Arc<IPCAgentProxy>,
+}
+
+impl PollingParentSyncer {
+    pub fn new(
+        config: Config,
+        parent_view_provider: Arc<InMemoryFinalityProvider>,
+        agent: Arc<IPCAgentProxy>,
+    ) -> Self {
+        Self {
+            config,
+            parent_view_provider,
+            agent,
+        }
+    }
+}
+
+impl PollingParentSyncer {
+    /// Start the parent finality listener in the background
+    pub fn start(self) -> anyhow::Result<()> {
+        let config = self.config;
+        let provider = self.parent_view_provider;
+        let agent = self.agent;
+
+        tokio::spawn(async move {
+            loop {
+                // Syncing with parent with the below steps:
+                // 1. Get the latest height in cache or latest height committed increment by 1 as the
+                //    starting height
+                // 2. Get the latest chain head height deduct away N blocks as the ending height
+                // 3. Fetches the data between starting and ending height
+                // 4. Update the data into cache
+                if let Err(e) = sync_with_parent(&config, &agent, &provider).await {
+                    tracing::error!("sync with parent encountered error: {e}");
+                }
+            }
+        });
+
+        Ok(())
+    }
+}
+
+async fn sync_with_parent<T: ParentFinalityProvider + Send + Sync + 'static>(
+    config: &Config,
+    agent_proxy: &Arc<IPCAgentProxy>,
+    provider: &Arc<T>,
+) -> anyhow::Result<()> {
+    let mut interval = tokio::time::interval(Duration::from_secs(config.polling_interval_secs));
+
+    loop {
+        interval.tick().await;
+
+        let starting_height = get_starting_height(provider).await?;
+        let latest_height = agent_proxy
+            .get_chain_head_height()
+            .await
+            .context("cannot fetch parent chain head")?;
+        if latest_height < config.chain_head_delay {
+            tracing::debug!("latest height not more than the chain head delay");
+            continue;
+        }
+        let ending_height = latest_height - config.chain_head_delay;
+
+        tracing::debug!("starting height: {starting_height}, ending_height: {ending_height}");
+
+        // we are going backwards in terms of block height, the latest block height is lower
+        // than our previously fetched head. It could be a chain reorg. We clear all the cache
+        // in `provider` and start from scratch
+        if starting_height > ending_height {
+            todo!()
+        }
+
+        let new_parent_views =
+            get_new_parent_views(agent_proxy, starting_height, ending_height).await?;
```

Comment:
Need to be somewhat careful here that if a node hasn't seen anything previously then it will I think start from the Genesis of Lotus and load everything into memory. 

The IPC genesis of Fendermint should have a non-zero starting height (the height at which this subnet has been created), and as we discussed earlier this routine must not be started while the node is syncing with others, so new joiners don't fetch the entire history of the subnet.

Reply:
There is an issue that tracks `node is syncing with others`. 
Regarding the genesis, I think `fendermint` alone might not be able to handle this, but instead should be handled by the gateway contract. If there are no previous committed finality, the contract should return the `genesis` finality with the `epoch` that the subnet was created and an initial block hash, like `vec![0; 32]`.